crowbar_framework/app/models/service_object.rb

```diff
@@ -1043,12 +1036,43 @@ def apply_role(role, inst, in_queue)
       @logger.debug "run_order #{run_order.inspect}"
     end
 
+    applying_nodes = run_order.flatten
+
+    # Prevent any intervallic runs from running whilst we apply the
+    # proposal, in order to avoid the orchestration problems described
+    # in https://bugzilla.suse.com/show_bug.cgi?id=857375
+    #
+    # First we pause the chef-client daemons by ensuring a magic
+    # pause-file.lock exists which the daemons will honour due to a
+    # custom patch:
+    nodes_to_lock = applying_nodes.reject do |node_name|
+      node = NodeObject.find_node_by_name(node_name)
+      node[:platform] == "windows" || node.admin?
+    end
+
+    apply_locks = nodes_to_lock.map do |node|
+      Crowbar::Lock::SharedNonBlocking.new(
+        logger: @logger,
+        path: "/var/chef/cache/pause-file.lock",
+        node: node,
+        owner: "apply_role-#{role.name}-#{inst}-#{Process.pid}",
+        reason: "apply_role(#{role.name}, #{inst}, #{in_queue}) pid #{Process.pid}"
+      ).acquire
+    end
+
+    # Now that we've ensured no new intervallic runs can be started,
+    # wait for any which started before we paused the daemons:
```

Comment:
Question on comment style, but usually, I think we just end the comment with `.`, not with `:` (since it's pretty clear the comment applies to what's below). Or am I the one doing this wrong? :-)